onnxruntime/core/providers/tensorrt/tensorrt_provider_factory.cc

```diff
@@ -84,11 +82,12 @@ struct Tensorrt_Provider : Provider {
     trt_options.trt_fp16_enable = internal_options.fp16_enable;
     trt_options.trt_int8_enable = internal_options.int8_enable;
 
-    if (internal_options.int8_calibration_table_name.size() == 0) {
+    auto str_size = internal_options.int8_calibration_table_name.size(); 
+    if (str_size == 0) {
       trt_options.trt_int8_calibration_table_name = nullptr;
     } else {
-      trt_options.trt_int8_calibration_table_name = new char[internal_options.int8_calibration_table_name.size() + 1];
-      strcpy((char*)trt_options.trt_int8_calibration_table_name, internal_options.int8_calibration_table_name.c_str());
+      trt_options.trt_int8_calibration_table_name = new char[str_size + 1];
+      strncpy((char*)trt_options.trt_int8_calibration_table_name, internal_options.int8_calibration_table_name.c_str(), str_size + 1);
```

Comment:
strncpy won't add the null char. I would change this to ```strncy(dest, src, str_size); and then dest[str_size] = '\0';```

Reply:
I was wondering why strncpy won't add the null char?
In this case, the size of dest string is always same as str_size, so strncpy will always copy the null char of dest and then stop due to str_size + 1 is reached.

https://en.cppreference.com/w/c/string/byte/strncpy

Reply:
I think it's better to be explict.

Reply:
Agree to be explicit in order to avoid null termination bug.